src/main/java/org/codehaus/mojo/versions/PluginUpdatesReport.java

```diff
@@ -27,31 +33,42 @@
 import org.apache.maven.plugins.annotations.ResolutionScope;
 import org.apache.maven.reporting.MavenReportException;
 import org.codehaus.mojo.versions.utils.PluginComparator;
-import org.codehaus.plexus.util.StringUtils;
 
-import java.io.File;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
-import java.util.TreeSet;
+import static org.codehaus.mojo.versions.utils.MiscUtils.filter;
 
 /**
  * Generates a report of available updates for the plugins of a project.
  *
  * @author Stephen Connolly
  * @since 1.0-beta-1
  */
-@Mojo( name = "plugin-updates-report", requiresProject = true, requiresDependencyResolution = ResolutionScope.RUNTIME, threadSafe = true )
-public class PluginUpdatesReport
-    extends AbstractVersionsReport
+@Mojo( name = "plugin-updates-report", requiresDependencyResolution = ResolutionScope.RUNTIME,
+       threadSafe = true )
+public class PluginUpdatesReport extends AbstractVersionsReport
 {
 
     /**
      * Report formats (html and/or xml). HTML by default.
-     * 
      */
     @Parameter( property = "pluginUpdatesReportFormats", defaultValue = "html" )
-    private String[] formats = new String[] { "html" };
+    private final String[] formats = new String[] { "html" };
```

Comment:
Don't use final for plugin parameters.
This field is set by Maven with reflection during plugin configuration.
With final modifier jvm tr can inline value.

Reply:
Got it.

Reply:
Applied review comments.